build-system/release-tagger/create-release.js

```diff
@@ -23,20 +23,113 @@
  */
 
 const argv = require('minimist')(process.argv.slice(2));
+const dedent = require('dedent');
 const {
   createRelease,
   getPullRequestsBetweenCommits,
   getRelease,
 } = require('./utils');
 const {GraphQlQueryResponseData} = require('@octokit/graphql'); //eslint-disable-line no-unused-vars
 
+/**
+ * Format pull request line
+ * @param {GraphQlQueryResponseData} pr
+ * @return {string}
+ */
+function _formatPullRequestLine(pr) {
+  const {mergeCommit, number, title, url} = pr;
+  const {abbreviatedOid, commitUrl} = mergeCommit;
+  return dedent`\
+  <a href="${commitUrl}"><code>${abbreviatedOid}</code></a> ${title} \
+  (<a href="${url}">#${number}</a>)`;
+}
+
+/**
+ * Create raw notes
+ * @param {Array<GraphQlQueryResponseData>} prs
+ * @return {Array<string>}
+ */
+function _createRawNotes(prs) {
+  return prs.map((pr) => _formatPullRequestLine(pr));
+}
+
+/**
+ * Organize pull requests into sections
+ * @param {Array<GraphQlQueryResponseData>} prs
+ * @return {Array<string>}
+ */
+function _createSections(prs) {
+  const sections = {
+    'ads': [],
+    'build-system': [],
+    'src': [],
+    'third_party': [],
+    'validator': [],
+    'package updates': new Array(), // [] defaults to never[], which tsc flags
+  };
+
+  for (const pr of prs) {
+    // renovate bot
+    if (pr.author.login === 'renovate-bot') {
+      sections['package updates'].push(_formatPullRequestLine(pr));
+      continue;
+    }
+
```

Comment:
I wonder if this could "look nicer" if you use a function style instead:
```javascript
  sections['package updates'] = prs
      .filter(pr => pr.author.login === 'renovate-bot')
      .map(_formatPullRequestLine));
```

Reply:
Logic here is weird, I need the `continue` to avoid adding renovate PRs under other sections i.e. build-system package updates.